src/main/java/htsjdk/samtools/util/AbstractProgressLogger.java

```diff
@@ -60,10 +61,17 @@ private synchronized void record() {
         if (this.lastChrom == null) readInfo = "*/*";
         else readInfo = this.lastChrom + ":" + fmt.format(this.lastPos);
 
+        final String rnInfo;
+        if(lastRN != null) {
+            rnInfo = ", last read name: " + lastRN;
+        } else {
+            rnInfo = "";
+        }
+
         final long n = (this.processed % this.n == 0) ? this.n : this.processed % this.n;
 
         log(this.verb, " ", processed, " " + noun + ".  Elapsed time: ", elapsed, "s.  Time for last ", fmt.format(n),
-                ": ", period, "s.  Last read position: ", readInfo);
+                ": ", period, "s.  Last read position: ", readInfo, rnInfo);
```

Comment:
I can make a protected alternate constructor with the bool explicit and/or a `setLogReadNames` method, but I'm trying not to mess with the `ProgressLoggerInterface` so it'd only be available to subclasses of `AbstractProgressLogger`.

IMO this should roll out by default and folks can turn it off if they specifically object. I'll let the hivemind chime in if there are objections

Reply:
perhaps the logger can figure out if the records are sorted or not, and only emit readnames when they are not?

Reply:
it doesn't do that for chr/pos

Reply:
I will also vote for default to a switch for this feature, and default to off (to keep the previous behaviour).

Reply:
@helgridly: I think that if you add the feature that the logger will output readname only after it notices that the records are not coordinate-sorted, this will be a good compromise between, providing information and terseness.

Reply:
providing a switch would require adding yet another Default setting to HTSJDK which is gross.

Reply:
Cannot be done on construction?